internal/pkg/cli/pipeline_list.go

```diff
@@ -27,43 +32,78 @@ import (
 const (
 	pipelineListAppNamePrompt = "Which application are the pipelines in?"
 	pipelineListAppNameHelper = "An application is a collection of related services."
+
+	pipelineListTimeout = 10 * time.Second
 )
 
 type listPipelineVars struct {
-	appName          string
-	shouldOutputJSON bool
+	appName                  string
+	shouldOutputJSON         bool
+	shouldShowLocalPipelines bool
 }
 
 type listPipelineOpts struct {
 	listPipelineVars
-	codepipeline   pipelineGetter
-	pipelineLister deployedPipelineLister
 	prompt         prompter
 	sel            configSelector
 	store          store
 	w              io.Writer
+	workspace      wsPipelineGetter
+	pipelineLister deployedPipelineLister
+
+	newDescriber newPipelineDescriberFunc
+
+	wsAppName string
 }
 
+type newPipelineDescriberFunc func(pipeline deploy.Pipeline) (describer, error)
+
 func newListPipelinesOpts(vars listPipelineVars) (*listPipelineOpts, error) {
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, err
+	}
+
 	defaultSession, err := sessions.ImmutableProvider(sessions.UserAgentExtras("pipeline ls")).Default()
 	if err != nil {
 		return nil, fmt.Errorf("default session: %w", err)
 	}
+
+	var wsAppName string
+	if vars.shouldShowLocalPipelines {
+		wsAppName = tryReadingAppName()
+		if vars.appName == "" {
+			vars.appName = wsAppName
+		}
+	}
+
 	store := config.NewSSMStore(identity.New(defaultSession), ssm.New(defaultSession), aws.StringValue(defaultSession.Config.Region))
 	prompter := prompt.New()
 	return &listPipelineOpts{
 		listPipelineVars: vars,
-		codepipeline:     codepipeline.New(defaultSession),
 		pipelineLister:   deploy.NewPipelineStore(rg.New(defaultSession)),
 		prompt:           prompter,
 		sel:              selector.NewConfigSelect(prompter, store),
 		store:            store,
 		w:                os.Stdout,
+		workspace:        ws,
+		newDescriber: func(pipeline deploy.Pipeline) (describer, error) {
+			return describe.NewPipelineDescriber(pipeline, false)
+		},
+		wsAppName: wsAppName,
 	}, nil
 }
 
 // Ask asks for and validates fields that are required but not passed in.
 func (o *listPipelineOpts) Ask() error {
+	if o.shouldShowLocalPipelines {
+		if err := validateInputApp(o.wsAppName, o.appName, o.store); err != nil {
```

Comment:
Do you think we can rename this function to something like `validateWorkspaceApp` so that it's clear that this function is used when the command needs to be run within a workspace?

Reply:
Also what do you think of moving this block to `Validate` since all it does is validating that:
1. There is a workspace
2. The workspace doesn't conflict with `--app` flag
3. The app of the workspace exists

without needing to ask anything.

Reply:
nit: we could probably just:
```suggestion
		return validateInputApp(o.wsAppName, o.appName, o.store)
```

Reply:
> Also what do you think of moving this block to Validate

I think since we'd have to have a similar block in `Ask()` _anyways_ (like this:
```
Validate() {
  if o.shouldShowLocal {
    return validateInputApp()
  }
}

Ask() {
  if o.shouldShowLocal {
    return nil
  }
  
  // other app name validation
}
```
) it makes sense to consider this block as part of the "app name" validation, and leave it in `Ask()`. Alternatively we could remove the check in ask, but then we'd call `store.GetApplication()` twice, since `validateInputApp()` already calls it.